File: Semestru 4/MPP/lab/.net/lab3/lab3/Server/service/ServiceLogin.cs
using Model.app.domain;
using Persistence.app.repo.@interface;
using Services.services;

namespace Server.app.service
{
	public class ServiceLogin : IServiceLogin
	{
		private ILoginRepository Repo;

		public ServiceLogin(ILoginRepository repo)
		{
			this.Repo = repo;
		}

		public LoginInfo GetByUsername(string username) =>
			this.Repo.GetByUsername(username);
	}
}

File: Semestru 4/MPP/lab/java/lab3/Services/src/main/java/org/example/services/ServiceLogin.java
package org.example.services;

import org.example.domain.LoginInfo;
import org.example.repository.interf.ILoginRepository;

public class ServiceLogin implements IServiceLogin{
    private ILoginRepository Repo;

    public ServiceLogin(ILoginRepository repo) {
        this.Repo = repo;
    }

    public LoginInfo GetByUsername (String username) {
        return this.Repo.GetByUsername(username);
    }
}
